### ISSUE TITLE
1.8 Release

### ISSUE THREAD

**Issue by azz** [prettier/prettier#3117]
Markdown was merged a little while ago and it's been almost a month since the last patch, so it's probably time for another release!

Things we should merge first:

- [x] `getSupportInfo()` #3033 
- [x]  CSS grid declaration #3133 
- [x] JSX bug #3154
- [x] JSX spread TS bug #3158

Airbnb items:

- [ ] Allow inline method chaining if line length does not exceed `printWidth` #3107
- [ ] Line break after object property #3108

Misc:

- [x] Release Notes (edit: now needs ~~#3124~~)
- ~~Uncomment new feature #3138 (added by @ikatyang)~~
- ~~Merge description for Markdown support #3028 (added by @ikatyang)~~
- [x] Move all docs to website (1.8 docs included) #3139 (added by @ikatyang)
- [x] Upgrade to `jest-docblock@21.3.0-beta.7` (https://github.com/facebook/jest/pull/4837)

@vjeux, @sharmilajesupaul, @ljharb: should we hold out for Airbnb compat, or leave that for the next release?

Collaborators, I'd appreciate people reviewing the release notes draft: https://github.com/prettier/prettier/releases/tag/untagged-98ba38a8a1d3dcdd0d68

**Comment by ikatyang** [comment]
I think we should add a note for CJK markdown users about the rendered content:

> NOTE for CJK users:
> 
> If your markdown renderer does not support CJK line ending, you'll have to use plugin like [markdown-it-perfect-newline-for-cjk](https://www.npmjs.com/package/markdown-it-perfect-newline-for-cjk), [hexo-filter-fix-cjk-spacing](https://www.npmjs.com/package/hexo-filter-fix-cjk-spacing), etc. to remove additional spaces.
> 
> (source)
> 
> ```
> 一二三
> 四五六
> 七八九
> ``` 
>
> (rendered content with unsupported renderer)
>
> ```
> 一二三 四五六 七八九
> ```
>
> (rendered content with supported renderer or via plugin)
>
> ```
> 一二三四五六七八九
> ```

---

For the markdown list:

> Note: you can actually opt out of this by using 1. for all list items if you want to optimize for cleaner diffs.

Would it be better to mention that the trigger is just the second number?

**Comment by azz** [comment]
Feel free to add the CJK note. 

R.e. lists: that's an implementation detail, and we don't want to be bound to it in case it needs tinkering.

**Comment by ikatyang** [comment]
Done! https://github.com/prettier/prettier/releases/edit/untagged-f5c05ee380fe2f76eefe

The release draft's url is varying between updates and we can't easily know the edit history, it'd be better to put it in a temp branch/PR I guess.

**Comment by ljharb** [comment]
@azz it would be awesome if you could get as much of our tweaks in as possible; however if you're willing to make a rapid release after 1.8, then there's no reason to wait :-D (version numbers are cheap)

**Comment by vjeux** [comment]
Awesome release notes!

For the Airbnb issues, I'd love to keep the discussions going so we end up with a good outcome. We fixed a handful of straight bugs (yay!) but the other ones are either bigger issues or would affect a lot of existing code, so I want to make sure that we end up with a good solution rather than rushing for the first one.

**Comment by vjeux** [comment]
@ikatyang you can see the latest draft here: https://github.com/prettier/prettier/releases

**Comment by duailibe** [comment]
@azz I added the notes regarding #3124 but please change wording it if you think it's necessary

**Comment by azz** [comment]
Looks good, thanks!

**Comment by azz** [comment]
Anyone want to run current master though some sizeable codebases before we do the release?

**Comment by azz** [comment]
Just ran `master` on a 3,000 TS file codebase formatted with 1.7.4... 0 changes.

**Comment by lydell** [comment]
Tried it on a 15k LOC project. The only changes were how template literals break:

```diff
-import(/* webpackChunkName: "views/[request]" */ `./views/${script}.entry.js`);
+import(/* webpackChunkName: "views/[request]" */ `./views/${
+  script
+}.entry.js`);
```

```diff
-const size = `${width > 0 ? width : '?'}×${height > 0
-  ? height
-  : '?'}\u00a0mm²`;
+const size = `${width > 0 ? width : '?'}×${
+  height > 0 ? height : '?'
+}\u00a0mm²`;
```

**Comment by duailibe** [comment]
80k LOC js app, lots of changes. Some are JSDoc and template literals, but also the following changes:

```diff
-  static fields = [
-    'description',
-    'duration',
-    'id',
-    'links',
-    'thumbnail',
-    'title',
-    'url',
-  ].map(name => ({ name }));
+  static fields = ['description', 'duration', 'id', 'links', 'thumbnail', 'title', 'url'].map(
+    name => ({ name })
+  );
```
```diff
-const timers = [
-  'setTimeout',
-  'setInterval',
-  'clearTimeout',
-  'clearInterval',
-].reduce((acc, name) => {
-  acc[name] = delegate(name);
-  return acc;
-}, {});
+const timers = ['setTimeout', 'setInterval', 'clearTimeout', 'clearInterval'].reduce(
+  (acc, name) => {
+    acc[name] = delegate(name);
+    return acc;
+  },
+  {}
+);
```
```diff
 const bagOfWords = text =>
-  text.split(' ').reduce((acc, word) => {
-    acc.total += 1;
-    acc.counter[word] = (acc.counter[word] || 0) + 1;
-    return acc;
-  },
-  { total: 0, counter: {} });
+  text.split(' ').reduce(
+    (acc, word) => {
+      acc.total += 1;
+      acc.counter[word] = (acc.counter[word] || 0) + 1;
+      return acc;
+    },
+    { total: 0, counter: {} }
+  );
```
```diff
-    const chapterProgressIcons = ((chapterSummary && chapterSummary.chapterProgresses) ||
-      []
-    ).map(cp => {
-      [REDACTED]
-    });
+    const chapterProgressIcons = ((chapterSummary && chapterSummary.chapterProgresses) || []).map(
+      cp => {
+        [REDACTED]
+      }
+    );
```

**Comment by vjeux** [comment]
One regression here: https://github.com/prettier/prettier/pull/3110#issuecomment-341864862

**Comment by vjeux** [comment]
@duailibe: I also see a bunch of those, it doesn't feel like an intentional change, I'm trying to bisect.

**Comment by existentialism** [comment]
Ran it on two of our apps (~23k / ~15k LOC), and similar to @lydell, all changes were template literal breaks.

**Comment by vjeux** [comment]
@duailibe: it's a regression from https://github.com/prettier/prettier/pull/3079#issuecomment-341865341

**Comment by duailibe** [comment]
@vjeux thanks for investigating!

**Comment by azz** [comment]
As I mentioned, I ran it on a massive codebase yesterday with no changes, which is odd because I would have expected some JDoc alignment fixes at the very least... I'll have to investigate.

**Comment by vjeux** [comment]
Once https://github.com/prettier/prettier/pull/3140 lands, it's good on my side. I went through tons of diffs and nothing stands out as being a regression.

**Comment by ikatyang** [comment]
#3082 should be the post-release item, at least the `parser: markdown` part.

**Comment by azz** [comment]
D'oh, I had my remotes meddled on my work PC, was running with my fork's `master` branch, explains no changes.

When using the correct build I had heaps of changes, most template string and factory pattern changes, some JSDoc alignments. All great.

Thoughts on these?

```diff
-            var colorSet = Enumerable.from(
-                this.availableColorSets()
-            ).firstOrDefault((info, index) => {
-                return info.key == setName;
-            }, this.availableColorSets[0]);
+            var colorSet = Enumerable.from(this.availableColorSets()).firstOrDefault(
+                (info, index) => {
+                    return info.key == setName;
+                },
+                this.availableColorSets[0]
+            );
```

```diff
-                const columnDefinition = GridColumnStore
-                    .for("MarginLender")
-                    .get("marginStatus");
+                const columnDefinition = GridColumnStore.for("MarginLender").get(
+                    "marginStatus"
+                );
```

```diff
-            this.redactedRedactedRedacted(
-                redacted
-            ).then((redacted: IRedacted<RedactedRedacted>) => {
-                this.redactedRedactedRedacted(redacted.data);
-            });
+            this.redactedRedactedRedacted(redacted).then(
+                (redacted: IRedacted<RedactedRedacted>) => {
+                    this.redactedRedactedRedacted(redacted.data);
+                }
+            );
```

While there were a lot of changes none of them look obviously worse.

**Comment by vjeux** [comment]
@azz: all those are from https://github.com/prettier/prettier/pull/3079#issuecomment-341865341

It's not intentional and triggers a lot of changes but I kinda like them. I wouldn't be opposed to shipping it.

**Comment by vjeux** [comment]
Would love to get #3143 in as well :p

**Comment by hawkrives** [comment]
I ran bb145c1 on [gobbldygook][gb] and [all about olaf][aao], my two projects. Overall, it looks good!

✅ I mostly see template strings breaking on interpolations, which is a bit strange but I think I can learn to like.

<details><summary>diff 1</summary>

```diff
# https://github.com/StoDevX/AAO-React-Native/compare/0c56db6...prettier-1.8#diff-d191c3dbd80b90dce4e96e560d35c126
 <details>
   <summary>:exclamation: Big PR!</summary>
   <blockquote>
-    <p>We like to try and keep PRs under ${bigPRThreshold} lines, and this one was ${thisPRSize} lines.</p>
+    <p>We like to try and keep PRs under ${
+      bigPRThreshold
+    } lines, and this one was ${thisPRSize} lines.</p>
     <p>If the PR contains multiple logical changes, splitting each change into a separate PR will allow a faster, easier, and more thorough review.</p>
   </blockquote>
 </details>`,
```
</details>
<details><summary>diff 2</summary>

```diff
# https://github.com/StoDevX/AAO-React-Native/compare/0c56db6...prettier-1.8#diff-1749a9e97a8cc282aa52f39a4c1b1905
-export const AAO_USER_AGENT = `AllAboutOlaf/${aaoVersion} (${platformString}/${platformVersion})`
+export const AAO_USER_AGENT = `AllAboutOlaf/${aaoVersion} (${platformString}/${
+  platformVersion
+})`
```
</details>
<details><summary>diff 3</summary>

```diff
# https://github.com/StoDevX/AAO-React-Native/compare/0c56db6...prettier-1.8#diff-c534d0a55d849662638866310565344d
@@ -41,7 +41,9 @@ export class GoogleCalendarView extends React.Component {
   }
 
   buildCalendarUrl(calendarId: string) {
-    let calendarUrl = `https://www.googleapis.com/calendar/v3/calendars/${calendarId}/events`
+    let calendarUrl = `https://www.googleapis.com/calendar/v3/calendars/${
+      calendarId
+    }/events`
     let params = {
       maxResults: 50,
       orderBy: 'startTime',
```
</details><br/>

❓ There were a few spots where it un-inlined a ternary in a template string interpolation.

<details><summary>diff 1</summary>

```diff
# https://github.com/StoDevX/AAO-React-Native/compare/0c56db6...prettier-1.8#diff-b38c0ee835a71c9ce39fea6ccf4cfbd4
@@ -41,9 +41,11 @@ export default function NewsItem(props: {
     <header class="aao-header">
       <h1>${story.title}</h1>
     </header>
-    ${embedFeaturedImage && story.featuredImage
-      ? `<img src="${story.featuredImage}">`
-      : ''}
+    ${
+      embedFeaturedImage && story.featuredImage
+        ? `<img src="${story.featuredImage}">`
+        : ''
+    }
     ${story.content}
   `
```
</details>
<details><summary>diff 2</summary>

```diff
# https://github.com/hawkrives/gobbldygook/compare/01d8e4e...prettier-1.8#diff-ae7d13facd7ddbc7abbac9219bf6313f
diff --git a/.webpackrc.js b/.webpackrc.js
index 5748610f..f853679a 100644
--- a/.webpackrc.js
+++ b/.webpackrc.js
@@ -188,16 +188,24 @@ function config() {
 
             <link rel="chrome-webstore-item" href="https://chrome.google.com/webstore/detail/nhhpgddphdimipafjfiggjnbbmcoklld">
 
-            ${isProduction
-                ? '<script src="https://d2wy8f7a9ursnm.cloudfront.net/bugsnag-3.min.js" data-apikey="7e393deddaeb885f5b140b4320ecef6b"></script>'
-                : ''}
-            ${isProduction
-                ? '<script src="https://cdn.polyfill.io/v2/polyfill.js"></script>'
-                : ''}
-
-            ${context.css
-                ? `<link rel="stylesheet" href="${publicPath}${context.css}">`
-                : ''}
+            ${
+                isProduction
+                    ? '<script src="https://d2wy8f7a9ursnm.cloudfront.net/bugsnag-3.min.js" data-apikey="7e393deddaeb885f5b140b4320ecef6b"></script>'
+                    : ''
+            }
+            ${
+                isProduction
+                    ? '<script src="https://cdn.polyfill.io/v2/polyfill.js"></script>'
+                    : ''
+            }
+
+            ${
+                context.css
+                    ? `<link rel="stylesheet" href="${publicPath}${
+                          context.css
+                      }">`
+                    : ''
+            }
 
             <body><main id="gobbldygook"></main></body>
```
</details><br/>

✅ I quite like the changes from https://github.com/prettier/prettier/pull/3079#issuecomment-341865341.

<details><summary>diff</summary>

```diff
# https://github.com/StoDevX/AAO-React-Native/compare/0c56db6...prettier-1.8#diff-4d0e143e68c6a0f1747780ef8311813c
  fetchData = async () => {
-    const responseData: StudentOrgType[] = await fetchJson(
-      orgsUrl,
-    ).catch(err => {
-      reportNetworkProblem(err)
-      this.setState(() => ({error: true}))
-      return []
-    })
+    const responseData: StudentOrgType[] = await fetchJson(orgsUrl).catch(
+      err => {
+        reportNetworkProblem(err)
+        this.setState(() => ({error: true}))
+        return []
+      },
+    )
```
</details>

---

There was one spot where prettier inlined the argument to `expect` and then un-inlined the argument to `.toEqual`, but that's because the line is now exactly 80 chars long. Nothing to see here! 😅  (I confirmed ([1]) that with tabWidth:2, the argument is inlined correctly.)

<details><summary>diff</summary>

```diff
# https://github.com/hawkrives/gobbldygook/compare/01d8e4e...prettier-1.8#diff-1dc975e296003b4d6ee507717d52a8ae
diff --git a/modules/examine-student/__tests__/get-matches-from-children.test.js b/modules/examine-student/__tests__/get-matches-from-children.test.js
index 5e35cfda..5e6f18e4 100644
--- a/modules/examine-student/__tests__/get-matches-from-children.test.js
+++ b/modules/examine-student/__tests__/get-matches-from-children.test.js
@@ -249,42 +249,42 @@ describe('getMatchesFromChildren', () => {
             },
         }
 
-        expect(
-            getMatchesFromChildren(requirement.result, requirement)
-        ).toEqual([
-            {
-                $type: 'course',
-                department: ['CSCI'],
-                gereqs: ['AQR'],
-                number: 121,
-                _result: true,
-            },
-            {
-                $type: 'course',
-                department: ['ASIAN'],
-                number: 130,
-                _result: true,
-            },
-            {
-                $type: 'course',
-                department: ['ASIAN'],
-                number: 275,
-                _result: true,
-            },
-            {
-                $type: 'course',
-                department: ['THEAT'],
-                number: 222,
-                _result: true,
-                year: 2014,
-            },
-            {
-                $type: 'course',
-                department: ['THEAT'],
-                number: 222,
-                _result: true,
-                year: 2015,
-            },
-        ])
+        expect(getMatchesFromChildren(requirement.result, requirement)).toEqual(
+            [
+                {
+                    $type: 'course',
+                    department: ['CSCI'],
+                    gereqs: ['AQR'],
+                    number: 121,
+                    _result: true,
+                },
+                {
+                    $type: 'course',
+                    department: ['ASIAN'],
+                    number: 130,
+                    _result: true,
+                },
+                {
+                    $type: 'course',
+                    department: ['ASIAN'],
+                    number: 275,
+                    _result: true,
+                },
+                {
+                    $type: 'course',
+                    department: ['THEAT'],
+                    number: 222,
+                    _result: true,
+                    year: 2014,
+                },
+                {
+                    $type: 'course',
+                    department: ['THEAT'],
+                    number: 222,
+                    _result: true,
+                    year: 2015,
+                },
+            ]
+        )
     })
 })
```
</details>

[gb]: https://github.com/hawkrives/gobbldygook/compare/01d8e4e...prettier-1.8
[aao]: https://github.com/StoDevX/AAO-React-Native/compare/0c56db6...prettier-1.8

[1]: https://github.com/hawkrives/gobbldygook/compare/01d8e4e...prettier-1.8-tabwidth2?w=1#diff-1dc975e296003b4d6ee507717d52a8ae

**Comment by azz** [comment]
I think we're in pretty good shape for the release.

**Comment by azz** [comment]
Actually, let's crack #3158 first.

**Comment by azz** [comment]
Released 1.8.0! :tada: https://github.com/prettier/prettier/releases/tag/1.8.0
